packages/grid/x-data-grid/src/hooks/features/editing/useGridCellEditing.ts

```diff
@@ -179,7 +179,11 @@ export const useGridCellEditing = (
         }
 
         if (reason) {
-          const newParams: GridCellEditStartParams = { ...params, reason, key: event.key };
+          const newParams: GridCellEditStartParams = {
+            ...params,
+            reason,
+            key: params.colDef.type === 'number' ? Number(event.key) : event.key,
+          };
```

Comment:
Shouldn't we put the logic for this later in the process? Would this be valid on backspace?

Reply:
The `colDef` was already available here and I didn't want to pass it along or load it using selectors down the tree. To me, backspace works fine, did you face some issue?